src/Symfony/Component/Validator/Constraints/TimestampValidator.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+class TimestampValidator extends ConstraintValidator
+{
+
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Timestamp){
+            throw new UnexpectedTypeException($constraint::class, self::class);
+        }
+
+        if (null === $value){
+            return;
+        }
+
+        if (!is_int($value)){
+            throw new UnexpectedValueException($value, 'int');
+        }
+
+        if (0 > $value){
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $value)
+                ->setInvalidValue($value)
+                ->setCode(Timestamp::INVALID_TIMESTAMP_ERROR)
+                ->addViolation();
+        }
+
+        // Not sure if there is value that can fall in this case
+        $date = \DateTimeImmutable::createFromFormat('U', $value);
+        if ($date === false){
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $value)
+                ->setInvalidValue($value)
+                ->setCode(Timestamp::INVALID_TIMESTAMP_ERROR)
+                ->addViolation();
+        }
+        $timestamp = $value;
+
+        if ($constraint->greaterThan && date_create($constraint->greaterThan)->getTimestamp() >= $timestamp){
```

Comment:
* Don't use `date_create()`. It's an alias for the `DateTime` constructor, so let's just call the constructor instead.
* Please use `DateTimeImmutable` instead of `DateTime`.
* `$constraint->greaterThan` might contain a malformed date string. You don't handle this case yet.